10_rest/alextmz/cmd/puppy-server/main_test.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/puppy/store"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainHappyPathSyncStore(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	args = []string{"-d", "../../test/valid-formatted-json.json", "-s", "sync"}
+
+	go main()
+	// wait for all main() output to be done
+	<-syncoutput
+
+	// test if main() text output is what we expect
+	got := buf.String()
+	want := `Starting puppyserver with options:
+file   = ../../test/valid-formatted-json.json
+port  = 7735
+store = sync
+Loaded 3 puppies.
+Printing puppy id 1: puppy.Puppy{ID:1, Breed:"Dogo", Colour:"White", Value:500}
+Printing puppy id 2: puppy.Puppy{ID:2, Breed:"Mastiff", Colour:"Brindle", Value:700}
+Printing puppy id 3: puppy.Puppy{ID:3, Breed:"Fila", Colour:"Golden", Value:900}
+`
+	assert.Equal(t, want, got)
+
+	// test if main() is really serving the HTTP we expect
+	//
+	// unfortunately, looks like travis-ci does not allow
+	// connections to 127.0.0.1: we get 'connect: connection refused'
```

Comment:
hmm this sounds strange.
I'm pretty sure I recall using [httptest.Server](https://golang.org/pkg/net/http/httptest/#Server) which essentially starts a webserver on localhost in tests executed on Travis.
Can _you_ use httptest.Server?